dozer-types/src/arrow_types/from_arrow.rs

```diff
@@ -165,7 +169,22 @@ pub fn map_arrow_to_dozer_type(dt: &DataType) -> Result<FieldType, FromArrowErro
         DataType::Binary | DataType::FixedSizeBinary(_) | DataType::LargeBinary => {
             Ok(FieldType::Binary)
         }
-        DataType::Utf8 => Ok(FieldType::String),
+        DataType::Utf8 => {
```

Comment:
Utf8 handling in `map_value_to_dozer_field` needs to be handled accordingly.

Given that this is a buggy implementation but no test fails, I think we need round trip unit test: `Record` -> `RecordBatch` -> `Record`. `Record` should contain all kinds of fields.

Reply:
valid points.

1. But for `map_value_to_dozer_field` there is no way for us to get the metadata to check whether it's for `Json` or `Utf8`

```rust
DataType::Utf8 => {
            let array: Option<&serde_json::Value> =
                column.as_any().downcast_ref::<serde_json::Value>();
            if let Some(r) = array {
                let s: DozerField = if r.is_null() {
                    DozerField::Null
                } else {
                    DozerField::Json(
                        serde_json_to_json_value(r.clone())
                            .map_err(|_| FieldTypeNotSupported(column_name.to_string()))
                            .unwrap(),
                    )
                };
                Ok(s)
            } else {
                Ok(DozerField::Null)
            }
        }
```
would this be enough?

2. for the `field_test_cases` we already have the `Json` type, `test_field_serialize_roundtrip()` and this has passed. I don't get what you mean `round trip unit test: Record -> RecordBatch -> Record. Record should contain all kinds of fields.`?

Reply:
added!

Reply:
I think we need a round trip unit test: `Record` -> `RecordBatch` -> `Record`